gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -344,12 +345,15 @@ public GoogleHadoopFileSystemBase() {
 
     if (GhfsStorageStatistics.class.isAssignableFrom(globalStats.getClass())) {
       storageStatistics = (GhfsStorageStatistics) globalStats;
+
     } else {
       logger.atWarning().log(
           "Encountered an error while registering to GlobalStorageStatistics. Some of the GCS connector metrics will not be reported to metrics sinks. globalStatsClassLoader=<%s>; classLoader=<%s>",
           globalStats.getClass().getClassLoader(), GhfsStorageStatistics.class.getClassLoader());
       storageStatistics = GhfsStorageStatistics.DUMMY_INSTANCE;
     }
+
+    GoogleCloudStorageEventBus.register(storageStatistics);
```

Comment:
Let us not rely on Eventbus to dedup the subscribers? Why not do the registration in GhfsStorageStatistics?

Reply:
Enclosed the subscriber methods in class and registered in GhfsStorageStatistics.